src/ui/components/Library/Team/View/Tests/TestListPanel/TestListPanel.tsx

```diff
@@ -0,0 +1,113 @@
+import { useContext } from "react";
+import { ContextMenuItem, useContextMenu } from "use-context-menu";
+
+import LibraryDropdownTrigger from "ui/components/Library/LibraryDropdownTrigger";
+import { LibrarySpinner } from "ui/components/Library/LibrarySpinner";
+
+import { TeamContext } from "../../../TeamContextRoot";
+import { FilterField } from "../../TestRuns/FilterField";
+import {
+  TimeFilterContext,
+  TimeFilterOptions,
+  withinTeamRetentionLimit,
+} from "../../TimeFilterContextRoot";
+import { TestContext } from "../TestContextRoot";
+import { TestList } from "./TestList";
+
+const timeFilterLabel: Record<TimeFilterOptions, string> = {
+  "two-week": "Last two weeks",
+  week: "Last 7 days",
+  day: "Last day",
+  hour: "Last hour",
+  month: "Last 30 days",
+};
+const sortLabel = {
+  failureRate: "Sort by failure rate",
+  flakyRate: "Sort by flaky rate",
+  alphabetical: "Sort alphabetically",
+};
+
+export function TestListPanel() {
+  const { team } = useContext(TeamContext);
+  const { filterByText, setFilterByText, filterByTextForDisplay, sortBy, setSortBy, testsLoading } =
+    useContext(TestContext);
+  const { filterByTime, setFilterByTime } = useContext(TimeFilterContext);
+
+  const {
+    contextMenu: contextMenuSortBy,
+    onContextMenu: onClickSortBy,
+    onKeyDown: onKeyDownSortBy,
+  } = useContextMenu(
+    <>
+      <ContextMenuItem dataTestId="failureRate" onSelect={() => setSortBy("failureRate")}>
+        Sort by failure rate
+      </ContextMenuItem>
+      <ContextMenuItem dataTestId="flakyRate" onSelect={() => setSortBy("flakyRate")}>
+        Sort by flaky rate
+      </ContextMenuItem>
+      <ContextMenuItem dataTestId="alphabetical" onSelect={() => setSortBy("alphabetical")}>
+        Sort alphabetically
+      </ContextMenuItem>
```

Comment:
This is really minor but I've been meaning to reuse `sortLabel` above instead of rewriting values here.